monkey/agent_plugins/credentials_collectors/chrome/src/windows_credentials_database_processor.py

```diff
@@ -1,15 +1,133 @@
-from pathlib import PurePath
-from typing import Sequence
+import logging
+import os
+import shutil
+import sqlite3
+import string
+import tempfile
+from pathlib import PureWindowsPath
+from typing import List, Optional, Sequence, Set, Tuple
 
-from common.credentials import Credentials
+from Crypto.Cipher import AES
+
+from common.credentials import Credentials, Password, Username
 from common.types import Event
+from common.utils.code_utils import secure_generate_random_string
 
+from .windows_utils import Win32CryptUnprotectData
+
+logger = logging.getLogger(__name__)
 
-class WindowsCredentialsDatabaseProcessor:
-    def __init__(self):
-        pass
 
+# TODO: use pathlib.Path, not str
+class WindowsCredentialsDatabaseProcessor:
     def __call__(
-        self, interrupt: Event, database_paths: Sequence[PurePath]
+        self, interrupt: Event, database_paths: Set[Tuple[PureWindowsPath, Optional[bytes]]]
     ) -> Sequence[Credentials]:
-        return []
+        credentials = []
+
+        # TODO: make interruptible
+        for database_path, master_key in database_paths:
+            # copy database before querying it to bypass lock errors
+            path = self._copy_db(database_path)
+            if path:
+                try:
+                    credentials.extend(self._export_credentials(path, master_key))
+                except Exception as err:
+                    logger.debug(f"{err}")
+
+                self._clean_file(path)
+
+        return [
+            # TODO: fix assumption that identity is a username, could be an email ID;
+            #       or just let it be, anything that uses a `Username` which is actually
+            #       an email ID will just fail and move on
```

Comment:
We have two identity types, email and username. We should not put email addresses in username identity objects.